src/extractor/MappingExtractor.ts

```diff
@@ -0,0 +1,101 @@
+import { getPath } from '../utils';
+
+export class MappingExtractor {
+  static process(
+    input: ProcessableStructureDefinition,
+    elements: ProcessableElementDefinition[],
+    fhir: fhirdefs.FHIRDefinitions
+  ): ExportableMapping[] {
+    const mappings =
+      input.mapping?.map(m => {
+        const mapping = new ExportableMapping(m.identity);
+        mapping.source = input.name;
+        if (m.name) mapping.title = m.name;
+        if (m.uri) mapping.target = m.uri;
+        if (m.comment) mapping.description = m.comment;
+        return mapping;
+      }) ?? [];
+    elements.forEach(element => {
+      this.extractRules(element, mappings);
+    });
+
+    // Filter out mappings on the parent - only include mappings new to the profile
+    // or mappings on the base definition with additional mapping rules
+    const parent = fhir.fishForFHIR(
+      input.baseDefinition,
+      utils.Type.Resource,
+      utils.Type.Type,
+      utils.Type.Profile,
+      utils.Type.Extension
+    );
+    const newItems = differenceWith(input.mapping, parent?.mapping, isEqual);
+    const newMappings = mappings.filter(mapping => newItems.some(i => i.identity === mapping.name));
+    const inheritedMappings = mappings.filter(mapping =>
+      parent?.mapping.some((i: fhirtypes.StructureDefinitionMapping) => i.identity === mapping.name)
+    );
+
+    // For each inherited mapping, look to see if there are any new rules
+    const changedInheritedMappings = inheritedMappings.filter(mapping => {
+      const changedIndexes: number[] = [];
+      let isNew = false;
+      mapping.rules.forEach((rule, i) => {
+        const element = fhirtypes.StructureDefinition.fromJSON(parent).findElementByPath(
+          rule.path,
+          fhir
+        );
+        // See if any mapping on the element matches the current rule we are looking for
+        // If one matches, this is not a new rule. If none match, this is new to the profile.
+        const elementMappingRules = element.mapping?.map((m, i) =>
+          this.processMappingRule(ProcessableElementDefinition.fromJSON(element), m, i)
+        );
+        if (!elementMappingRules.find(r => isEqual(r, rule))) {
+          isNew = true;
+        } else {
+          // Remove inherited rules so we don't include them if the mapping is exported
+          changedIndexes.push(i);
+        }
+      });
+      pullAt(mapping.rules, changedIndexes);
+      return isNew;
+    });
+    return [...newMappings, ...changedInheritedMappings];
+  }
+
+  static extractRules(element: ProcessableElementDefinition, mappings: ExportableMapping[]) {
+    element.mapping?.forEach((mapping, i) => {
+      // Mappings are created at SD, so should always find a match at this point
+      const matchingMapping = mappings.find(m => m.name === mapping.identity);
+      if (!matchingMapping) return;
+      const mappingRule = this.processMappingRule(element, mapping, i);
+      matchingMapping.rules.push(mappingRule);
```

Comment:
Minor style request: instead of an early exit with a one-line if, this could be written as:
```suggestion
      if (matchingMapping) {
        const mappingRule = this.processMappingRule(element, mapping, i);
        matchingMapping.rules.push(mappingRule);
      }
```
I feel like this is a bit clearer at expressing "if there is a matching mapping, we make a `MappingRule` for it."